enhancements/okd/okd-featureset.md

```diff
@@ -0,0 +1,642 @@
+---
+title: okd-featureset
+authors:
+  - "@jatinsu"
+reviewers:
+  - "@JoelSpeed, for API review and feature gate validation"
+  - "@Prashanth684, for OKD-specific implementation details"
+  - "@cheesesashimi, for machine config implications"
+  - "@sdodson, for OKD platform considerations"
+approvers:
+  - "@sdodson"
+api-approvers:
+  - "@deads2k"
+creation-date: 2025-12-02
+last-updated: 2025-12-02
+tracking-link:
+  - https://issues.redhat.com/browse/OKD-259
+see-also:
+  - "/enhancements/installer/feature-sets.md"
+  - "/enhancements/okd-on-fedora-coreos-in-prow.md"
+replaces:
+superseded-by:
+---
+
+# OKD Feature Set
+
+## Summary
+
+This enhancement proposes the introduction of a new "OKD" feature set in the OpenShift API that will be enabled by default on all OKD clusters. This feature set allows OKD clusters to enable and test TechPreview features while maintaining the ability to upgrade, differentiating the community OKD distribution from the supported OpenShift Container Platform product.
+
+## Motivation
+
+OKD (OpenShift Kubernetes Distribution) is the community distribution of Kubernetes that powers OpenShift. Currently, OKD clusters use the same feature gates as OpenShift, which creates several challenges:
+
+1. OKD users cannot test TechPreview features without blocking upgrades (TechPreviewNoUpgrade)
+2. There is no mechanism to distinguish OKD-specific features from OpenShift features
+3. OKD's role as a testing ground for new features is limited by the same constraints as production OpenShift
+4. The community cannot experiment with features that may eventually graduate to OpenShift without losing upgrade capability
+
+By introducing an OKD-specific feature set, we enable the OKD community to test and provide feedback on new features while maintaining a stable upgrade path.
+
+### User Stories
+
+As an OKD cluster administrator, I want to enable TechPreview features on my cluster so that I can test upcoming OpenShift functionality and provide community feedback, without permanently blocking my ability to upgrade.
```

Comment:
A use case from the community's perspective is being able to adopt new features earlier in their lifecycle than OCP.